src/radical/pilot/agent/agent_0.py

```diff
@@ -340,6 +347,12 @@ def _start_services(self):
         If a `./services` file exist, reserve a compute node and run that file
         there as bash script.
         '''
+        # TODO
+        #  there is no pilot description/pilot instance accessible here. how to get services list
+        agent_service_tasks = [how to get services TaskDescription]
+        self.advance(agent_service_tasks, publish=False, push=True)
```

Comment:
You need to be careful: what you get from the client is a `TaskDescription` dictionary, what you want to `advance` here is a `Task` dictionary.  So you need to create a `Task` dictionary from the descriptions.  Look into the Raptor master for an example on how this can be done: [here](https://github.com/radical-cybertools/radical.pilot/blob/devel/src/radical/pilot/raptor/master.py#L325)

Reply:
tried. See now. @andre-merzky